src/app/(app)/product-list.tsx

```diff
@@ -1,94 +1,111 @@
+import { useEffect, useState } from 'react';
 
+import type { Product } from '@/api/products/types';
+import { useProducts } from '@/api/products/use-products';
 import { HeaderLogo } from '@/components/header-logo';
-import type { ProductCardProps } from '@/components/products-list/product-card';
-import { ProductCard } from '@/components/products-list/product-card';
+import { ProductListScrollView } from '@/components/products-list/products-scroll-view';
 import { SearchBar } from '@/components/search-bar';
-import {
-  FocusAwareStatusBar,
-  Image,
-  SafeAreaView,
-  ScrollView,
-  Text,
-  TouchableOpacity,
-  View,
-} from '@/ui';
+import { Image, SafeAreaView, Text, TouchableOpacity, View } from '@/ui';
 
-const products: ProductCardProps[] = [
-  {
-    id: '1',
-    name: 'Baumler Company Wood Chair 1',
-    price: 36,
-    status: 'Restored',
-  },
-  {
-    id: '2',
-    name: 'Baumler Company Wood Chair 2',
-    price: 36,
-    status: 'Restored',
-  },
-  {
-    id: '3',
-    name: 'Baumler Company Wood Chair 3',
-    price: 36,
-    status: 'Restored',
-  },
-  {
-    id: '4',
-    name: 'Baumler Company Wood Chair',
-    price: 36,
-    status: 'Restored',
-  },
-  {
-    id: '5',
-    name: 'Baumler Company Wood Chair',
-    price: 36,
-    status: 'Restored',
-  },
-];
+const FiltersButton = ({
+  products,
+  query,
+}: {
+  products: Product[];
+  query: string;
+}) => {
+  if (products.length === 0) {
+    return;
+  }
+  return (
+    <View
+      className={`absolute ${
+        query === '' ? 'bottom-20' : 'bottom-28'
+      } flex w-auto items-center justify-center self-center pb-20`}
+    >
+      <TouchableOpacity className="flex-row items-center gap-4 rounded-full bg-dark_violet p-4 px-8">
+        <Text className="text-lg font-bold text-white">Filers</Text>
+        <Image className="h-4 w-4" source={images.filterIcon()} />
+      </TouchableOpacity>
+    </View>
+  );
+};
+
+const SearchResult = ({
+  query,
+  clearQuery,
+}: {
+  query: string;
+  clearQuery: () => void;
+}) => {
+  if (query) {
+    return (
+      <View className="flex-row justify-between px-5 pt-6">
+        <Text className="text-lg text-gray-600">
+          You searched for "{query}”
+        </Text>
+        <TouchableOpacity onPress={clearQuery}>
+          <Text className="text-lg text-link">Clear all</Text>
+        </TouchableOpacity>
+      </View>
+    );
+  }
+};
 
 export default function ProducList() {
+  const [page, setPage] = useState(1);
+  const { data } = useProducts({ variables: { page: 1, items: 100 } });
+  const [products, setProducts] = useState<Product[]>([]);
+  const [query, setQuery] = useState<string>('');
+  const [filteredProducts, setFilteredProducts] = useState<Product[]>([]);
+  const [shouldReset, setShouldReset] = useState(false);
+
+  useEffect(() => {
+    const productList = query === '' ? data?.data ?? [] : filteredProducts;
+    setProducts(productList?.slice((page - 1) * 7, page * 7));
```

Comment:
Two things:
* I'm wondering if the `useProducts` has the variable `items`, wouldn't that solve that having to slice the `products` array every time, and just passing `items = 7`, and on each page, the API will return 7 products each time?
* Perhaps you can initialize the product state directly with the data that comes from the query, e.g:

```javascript
 ...query
  const [filteredProducts, setFilteredProducts] = useState<Product[]>([]);
  const productList = query === '' ? data?.data ?? [] : filteredProducts;
  const [products, setProducts] = useState<Product[]>(productList);
```
Once, you use setFilteredProduct and the query is not an empty string, `products` should shift to filteredProducts.

Again, this is a non-blocker thing, since this involves a refactor and could impact the behavior of the entire flow.